### ISSUE TITLE
An error occurred while Rails/Blank cop was inspecting

### ISSUE THREAD

**Issue by pschambacher** [bbatsov/rubocop#4210]
## Steps to reproduce the problem
```ruby
# frozen_string_literal: true

class User < ApplicationRecord
  class << self
    def create_or_update_from_hash(hash)
      find_by(external_id: hash[:external_id].to_s) || new
    end
  end
end
```

## Stack trace
```
An error occurred while Rails/Blank cop was inspecting /Users/pschambacher/Code/zendesk/sauron/app/models/user.rb:27:13.
undefined method `type' for nil:NilClass
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cop/rails/blank.rb:45:in `nil_or_empty?'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cop/rails/blank.rb:79:in `on_or'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cop/commissioner.rb:43:in `block (2 levels) in on_or'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cop/commissioner.rb:106:in `with_cop_error_handling'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cop/commissioner.rb:42:in `block in on_or'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cop/commissioner.rb:41:in `each'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cop/commissioner.rb:41:in `on_or'
(eval):3:in `on_lvasgn'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cop/commissioner.rb:47:in `on_lvasgn'
(eval):2:in `block in on_begin'
(eval):2:in `each'
(eval):2:in `on_begin'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cop/commissioner.rb:47:in `on_begin'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/ast/traversal.rb:95:in `on_def'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cop/commissioner.rb:47:in `on_def'
(eval):2:in `block in on_begin'
(eval):2:in `each'
(eval):2:in `on_begin'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cop/commissioner.rb:47:in `on_begin'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/ast/traversal.rb:141:in `on_while'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cop/commissioner.rb:47:in `on_sclass'
(eval):2:in `block in on_begin'
(eval):2:in `each'
(eval):2:in `on_begin'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cop/commissioner.rb:47:in `on_begin'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/ast/traversal.rb:88:in `on_class'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cop/commissioner.rb:47:in `on_class'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/ast/traversal.rb:12:in `walk'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cop/commissioner.rb:60:in `investigate'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cop/team.rb:124:in `investigate'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cop/team.rb:112:in `offenses'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cop/team.rb:54:in `inspect_file'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/runner.rb:248:in `inspect_file'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/runner.rb:195:in `block in do_inspection_loop'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/runner.rb:227:in `block in iterate_until_no_changes'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/runner.rb:220:in `loop'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/runner.rb:220:in `iterate_until_no_changes'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/runner.rb:191:in `do_inspection_loop'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/runner.rb:101:in `block in file_offenses'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/runner.rb:111:in `file_offense_cache'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/runner.rb:99:in `file_offenses'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/runner.rb:90:in `process_file'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/runner.rb:68:in `block in each_inspected_file'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/runner.rb:65:in `each'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/runner.rb:65:in `reduce'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/runner.rb:65:in `each_inspected_file'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/runner.rb:57:in `inspect_files'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/runner.rb:36:in `run'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cli.rb:72:in `execute_runner'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/lib/rubocop/cli.rb:27:in `run'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/bin/rubocop:13:in `block in <top (required)>'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/2.3.0/benchmark.rb:308:in `realtime'
/Users/pschambacher/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rubocop-0.48.0/bin/rubocop:12:in `<top (required)>'
/Users/pschambacher/.rbenv/versions/2.3.3/bin/rubocop:22:in `load'
/Users/pschambacher/.rbenv/versions/2.3.3/bin/rubocop:22:in `<main>'
```

## RuboCop version

```
$ rubocop -V
0.48.0 (using Parser 2.4.0.0, running on ruby 2.3.3 x86_64-darwin16)
```

**Comment by 4ndypanda** [comment]
Sweet I was going to post this error too. It seems to only occur when you use the '||' operator and the first expression is a function.

## Fails 
```ruby
def one_or_two?(num)
  one?(num) || num == 2
end

def one?(num)
  num == 1
end
```

## Passes 
```ruby
def one_or_two?(num)
  num == 2 || one?(num)
end

def one?(num)
  num == 1
end
```

**Comment by rrosenblum** [comment]
Thanks for the bug report. This is a duplicate of #4179. It has been fixed in #4190.
